tmt/base.py

```diff
@@ -1883,9 +1921,11 @@ def plans(self, keys=None, names=None, filters=None, conditions=None,
             sources = None
 
         # Build the list, convert to objects, sort and filter
-        plans = [
-            Plan(node=plan, run=run) for plan
-            in self.tree.prune(keys=keys, names=names, sources=sources)]
+        plans = [Plan(node=plan, run=run)
```

Comment:
I guess here would be the best place to decide whether we continue with just the dumb reference (e.g. used for `tmt plan show` so that it is not slow) or whether the remote plans should be fetched and their complete data used to initialize the `Plan` objects in full. The `Tree.plans()` method could have an additional parameter `deep=False` to choose whether a shallow or deep remote plan inspection should be done. Does it make sense?

Reply:
The deep/shallow functionality is being discussed in another conversation already, resolving this duplicate.